state_machine/applications/flight/Tasks/deployment_manager.py

```diff
@@ -1,21 +1,41 @@
 from lib.template_task import Task
 from pycubed import cubesat
 from state_machine import state_machine
+import time
 
 
+INITIAL_BURN_DELAY = 5 * 60
+BURN_INTERVAL = 24 * 60 * 60
+
 class deployment_manager(Task):
     name = 'depoloyment_manager'
     color = 'blue'
 
     rgb_on = False
+    start_time = time.time()
+    last_burn = float('inf')
 
     async def main_task(self):
         if (cubesat.f_contact):
             self.debug('Contact with ground station has been already established, switching to Normal mode')
             state_machine.switch_to('Normal')
-        else:
-            if await cubesat.burn(duration=15):
+        elif self.should_burn():
+            self.last_burn = time.time()
+            self.start_time = float('inf')
```

Comment:
What is the purpose of this?

Reply:
start_time set to infinity to make it so that time.time()-start_time is never greater than 5 minutes or whatever our constant is. We can consider setting a boolean or something instead.

Reply:
but why is it getting set to `inf` here? Wouldn't this overwrite the value it was initialized to on line 15?

Is this to prevent the burn from occuring during testing? If that's the case it definitely needs to be a well-named boolean.

Reply:
clearly the code isn't clear, I have an idea for how to clean it up.